MastodonSDK/Sources/MastodonUI/Helper/MediaLayoutHelper.swift

```diff
@@ -0,0 +1,314 @@
+//
+//  MediaLayoutHelper.swift
+//  
+//
+//  Created by Grishka on 25.03.2023.
+//
+
+import Foundation
+import MastodonSDK
+import CoreDataStack
+
+public struct MediaLayoutResult {
+	let width: Int
+	let height: Int
+	let columnSizes: [Int]
+	let rowSizes: [Int]
+	let tiles: [Tile]
+	
+	public struct Tile {
+		var colSpan: Int
+		let rowSpan: Int
+		var startCol: Int
+		let startRow: Int
+		var width: Int = 0
+	}
+}
+
+class MediaLayoutHelper {
+	static let maxWidth: Float = 1000
```

Comment:
Most of the rest of the code uses `CGFloat` (aka double) because that’s what the UIKit layout APIs accept.